src/VbsEnclaveSDK/src/veil_host_lib/hello_winrt_kcm.vtl0.cpp

```diff
@@ -0,0 +1,430 @@
+{
+    static constexpr wchar_t c_formatString[] = L"{}//{}//{}";
+    wil::unique_handle tokenHandle;
+
+    THROW_LAST_ERROR_IF(!OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY, &tokenHandle));
+
+    DWORD tokenInfoLength = 0;
+    GetTokenInformation(tokenHandle.get(), TokenUser, nullptr, 0, &tokenInfoLength);
+    std::vector<BYTE> tokenInfoBuffer(tokenInfoLength);
+
+    THROW_LAST_ERROR_IF(!GetTokenInformation(tokenHandle.get(), TokenUser, tokenInfoBuffer.data(), tokenInfoLength, &tokenInfoLength));
+
+    PTOKEN_USER tokenUser = reinterpret_cast<PTOKEN_USER>(tokenInfoBuffer.data());
+
+    // Extract the SID from the TOKEN_USER structure
+    PSID userSid = tokenUser->User.Sid;  // This is how you get the SID
+
+    // Convert SID to string
+    wil::unique_hlocal_ptr<WCHAR[]> userSidString;
+    THROW_LAST_ERROR_IF(!ConvertSidToStringSidW(userSid, wil::out_param_ptr<LPWSTR*>(userSidString)));
+
+    // Create the formatted key name
+    std::wstring result = std::format(c_formatString, userSidString.get(), userSidString.get(), name);
+
+    return result;
+}
+
+winrt::hstring GetAlgorithm(uintptr_t ecdhAlgorithm)
+{
+    using namespace winrt::Windows::Security::Cryptography::Certificates;
+    if (reinterpret_cast<BCRYPT_ALG_HANDLE>(ecdhAlgorithm) == BCRYPT_ECDH_P384_ALG_HANDLE)
+    {
+        return KeyAlgorithmNames::Ecdh384();
+    }
+    else if (reinterpret_cast<BCRYPT_ALG_HANDLE>(ecdhAlgorithm) == BCRYPT_ECDH_P256_ALG_HANDLE)
+    {
+        return KeyAlgorithmNames::Ecdh256();
+    }
+    THROW_HR(E_INVALIDARG);
+}
+
+// Helper function to convert veil_abi::Types::keyCredentialCacheConfig to KeyCredentialCacheConfiguration
+KeyCredentialCacheConfiguration ConvertCacheConfig(const veil_abi::Types::keyCredentialCacheConfig& cacheConfig)
+{
+    // Map cacheOption to KeyCredentialCacheOption
+    KeyCredentialCacheOption cacheOption;
+    switch (cacheConfig.cacheOption)
+    {
+        case 0:
+            cacheOption = KeyCredentialCacheOption::NoCache;
+            break;
+        case 1:
+            cacheOption = KeyCredentialCacheOption::CacheWhenUnlocked;
+            break;
+        default:
+            cacheOption = KeyCredentialCacheOption::NoCache; // Default fallback
+            break;
+    }
+
+    // Convert timeout from seconds to TimeSpan (100-nanosecond units)
+    winrt::Windows::Foundation::TimeSpan timeout{ static_cast<int64_t>(cacheConfig.cacheTimeoutInSeconds) * 10000000LL };
+
+    // Use WinRT's get_activation_factory for a more modern approach
+    auto factory = winrt::get_activation_factory<winrt::Windows::Security::Credentials::KeyCredentialCacheConfiguration, 
+                                                 winrt::Windows::Security::Credentials::IKeyCredentialCacheConfigurationFactory>();
+
+    return factory.CreateInstance(cacheOption, timeout, cacheConfig.cacheUsageCount);
+}
+
+// Helper function to validate that a COM object is still valid
+bool IsValidCOMObject(void* abi)
```

Comment:
>IsValidCOMObject [](http://example.com/codeflow?start=5&length=16)

I don't know if this function provides value.

Answering the question of whether or not an arbitrary void* is a com pointer or not - when the chain-of-custody of said pointer is entirely within our jurisdiction - doesn't provide any meaning, afaict.

Further, it's UB to cast a void* to a pointer (unless you're already certain it's a COM pointer) and call a COM virtual function on it.